src/Workspaces/SharedUtilitiesAndExtensions/Compiler/CSharp/Formatting/Engine/Trivia/CSharpTriviaFormatter.cs

```diff
@@ -99,6 +99,38 @@ protected override LineColumnRule GetLineColumnRuleBetween(SyntaxTrivia trivia1,
 
                 if (trivia2.IsKind(SyntaxKind.RegionDirectiveTrivia) || trivia2.IsKind(SyntaxKind.EndRegionDirectiveTrivia))
                 {
+                    // When we have a '#region' in conditionally disabled conditional (e.g, `#if false`), we cannot determine a correct indentation for '#region'.
+                    // So we preserve the existing indentation.
+                    // To figure whether we are in a disabled region, we do the following:
+                    // - Starting from the given trivia, keep going back.
+                    // - Once we find a disabled text, we know this is a disabled region.
+                    // - If we find a BranchingDirectiveTriviaSyntax, we can directly determine whether it's active or not via BranchTaken property.
+                    var previous = trivia2;
+                    while ((previous = previous.GetPreviousTrivia(previous.SyntaxTree, cancellationToken)) != default)
+                    {
+                        if (previous.IsKind(SyntaxKind.DisabledTextTrivia))
+                        {
+                            return LineColumnRule.Preserve;
+                        }
+                        else if (previous.IsKind(SyntaxKind.EndIfDirectiveTrivia))
+                        {
+                            // We cannot go back further to determine if we are in a disabled region or not.
+                            // We assume we are in active region, but that might not be correct.
+                            break;
```

Comment:
i'm curious why you can't go further back?  can't this go to the corresponding `#if` and continue searching?

Reply:
Ah, it's possible, something like this (pseudo-code):

```csharp
if (found #endif)
{
    int endifCount = 1;
    int ifCount = 0;
    while (go back)
    {
        if (found #endif) endifCount++;
        else if (found #if) ifCount++;
        if (ifCount == endifCount) break;
    }
}
```

It's a niche scenario though, and we might have to go back too much.

Reply:
If it's important to handle this, I'll go ahead and do it.